drivers/shared/src/get_parameters.c

```diff
@@ -563,3 +563,417 @@ get_parameters(FILE *paramfile)
         fgets(cmdstr, MAXSTRING, paramfile);
     }
 }
+
+/******************************************************************************
+ * @brief    Validate VIC model parameter values
+ *****************************************************************************/
+void
+validate_parameters()
+{
+
+    extern parameters_struct param;
+
+    // Validate Parameters
+    // Lapse Rate
+    if (param.LAPSE_RATE > 0.) {
+        log_err("Invalid value for LAPSE_RATE, must be <= 0.  "
+                "Units are degC/m");
+    }
+    if (param.LAPSE_RATE < -1) {
```

Comment:
I think these may be easier to read if they are collapsed into one check (and error message) per parameters.

E.g.:

```C
if (!(param.LAPSE_RATE  >= -1 && param.LAPSE_RATE <= 0)) {
    log_err("LAPSE_RATE must be defined on the interval [-1,0] (C/m)")
}
```